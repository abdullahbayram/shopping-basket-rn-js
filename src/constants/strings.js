export default {
  icons: {
    cardholderName: 'account',
    cardholderNameCheck: 'account-check',
    expirationDate: 'calendar-alert',
    expirationDateCheck: 'calendar-check',
    cvv: 'shield-alert',
    cvvCheck: 'shield-check',
    creditCardNumber: 'credit-card',
    creditCardNumberCheck: 'credit-card-check',
  },
  screens: {
    success: 'Success',
    payment: 'Payment',
    error: 'Error',
    checkout: 'Checkout',
  },
  devErrors: {
    parseErrorMessage: 'Failed to parse error message:',
  },
  buttons: {
    retry: 'Retry',
    checkout: 'CHECKOUT',
    order: 'ORDER',
    payAndOrder: 'PAY AND ORDER',
    gotoProducts: 'Go to Products',
  },
  productList: {
    basketItemCount: 'Items in the basket: ',
    loading: 'Loading products...',
    errorLoading: 'Unable to load products. Please try again.',
    limitReached: 'You reached the max quantity per product!',
  },
  checkout: {
    total: 'Total:',
    order: 'ORDER',
    promoCodePlaceholder: 'Enter your promo code',
    promoCode: 'Promo Code',
    promoCodeRequired: 'Promo code can not be empty',
    promoCodeNotValid: 'This promo code is not valid',
    applyPromo: 'APPLY PROMO CODE',
    emptyBasket: 'Your basket is empty.',
  },
  payment: {
    total: 'Total:',
    basketItemCount: 'Items in the basket: ',
    cardholderName: 'Cardholder Name',
    cardholderNamePlaceholder: 'Enter the name on your card',
    creditCardNumber: 'Credit Card Number',
    creditCardPlaceholder: 'Enter your credit card number',
    expirationDate: 'Expiration Date',
    expirationDatePlaceholder: 'MM/YY',
    cvv: 'CVV',
    cvvPlaceholder: '123',
    unexpectedError: 'An unexpected error occurred. Please try again later.',
    unexpectedErrorShort: 'Unexpected Error',
    invalidCard: 'Invalid credit card number',
    cardHolderRequired: 'Cardholder name is required',
    creditCardRequired: 'Credit card number is required',
    expirationDateRequired: 'Expiration date is required',
    cvvRequired: 'CVV is required',
    invalidExpirationDate: 'Invalid expiration date (MM/YY)',
    cvvLength: 'CVV must be 3 digits',
    cardHolderMinLength: 'Cardholder name must be at least 3 characters',
    success: `
Thank you!
Your order has been placed successfully.
      `,
  },
};
