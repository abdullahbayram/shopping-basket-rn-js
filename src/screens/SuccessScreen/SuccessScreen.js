import React, { useEffect, useRef, useState } from 'react';
import { StyleSheet, View, Animated, Text, BackHandler } from 'react-native';

const SuccessScreen = ({ navigation }) => {
  const [seconds, setSeconds] = useState(5);
  const progress = useRef(new Animated.Value(0)).current;

  useEffect(() => {
    // BackHandler: Disable Android back button
    const backHandler = BackHandler.addEventListener('hardwareBackPress', () => true);

    Animated.timing(progress, {
      toValue: 1,
      duration: 5000,
      useNativeDriver: false, // Required for width animations
    }).start();

    const interval = setInterval(() => {
      setSeconds((prev) => {
        if (prev === 1) {
          clearInterval(interval); // Clear interval when countdown ends
        }
        return prev - 1;
      });
    }, 1000);

    const timer = setTimeout(() => {
      navigation.reset({
        index: 0,
        routes: [{ name: 'Product List' }],
      });
    }, 5000);

    return () => {
      backHandler.remove();
      clearInterval(interval);
      clearTimeout(timer);
    };
  }, [navigation, progress]);

  const widthInterpolated = progress.interpolate({
    inputRange: [0, 1],
    outputRange: ['0%', '100%'], // Width from 0% to 100%
  });

  return (
    <View style={styles.container}>
      <View style={styles.textContainer}>
        <Text style={styles.title}>Thank you!</Text>
        <Text style={styles.title}>Your order has been placed successfully.</Text>
        <Text style={styles.message}>
          Redirecting to product list in {seconds} second{seconds > 1 ? 's' : ''}...
        </Text>
      </View>
      {/* Progress bar container */}
      <View style={styles.progressBarContainer}>
        <Animated.View style={[styles.progressBar, { width: widthInterpolated }]} />
      </View>
    </View>
  );
};

export default SuccessScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#fff',
  },
  textContainer: {
    marginBottom: 20,
    alignItems: 'center',
  },
  title: {
    fontWeight: 'bold',
    fontSize: 24,
    marginBottom: 8,
    textAlign: 'center',
  },
  message: {
    fontSize: 16,
    color: '#666',
  },
  progressBarContainer: {
    width: '80%',
    height: 10,
    backgroundColor: '#e0e0e0',
    borderRadius: 5,
    overflow: 'hidden', // Ensure the progress stays inside the container
  },
  progressBar: {
    height: '100%',
    backgroundColor: '#76c7c0',
    borderRadius: 5,
  },
});
