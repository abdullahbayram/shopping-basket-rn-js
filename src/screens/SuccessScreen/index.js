import React from 'react';
import { BaseScreen, RedirectWithAnimation } from '../../components/templetes';
import { strings } from '../../constants';

const SuccessScreen = () => {
  return (
    <BaseScreen>
      <RedirectWithAnimation message={strings.payment.success} />
    </BaseScreen>
  );
};

export default SuccessScreen;
