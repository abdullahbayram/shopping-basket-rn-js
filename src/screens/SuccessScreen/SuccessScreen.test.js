import React from 'react';
import { screen } from '@testing-library/react-native';
import SuccessScreen from './SuccessScreen';
import renderInNavigation from '../../../__tests__/utils/renderInNavigation';
import strings from '../../constants/strings';

describe('SuccessScreen Component', () => {
  it('should render the success message', () => {
    renderInNavigation(<SuccessScreen />);
    expect(screen.getByText(strings.payment.success)).toBeTruthy();
  });

  it('should match the snapshot', () => {
    const { toJSON } = renderInNavigation(<SuccessScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
});
