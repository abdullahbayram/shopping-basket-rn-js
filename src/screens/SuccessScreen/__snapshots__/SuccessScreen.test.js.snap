// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessScreen Component should match the snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(255, 251, 254, 1)",
      "flex": 1,
      "paddingLeft": 16,
      "paddingRight": 9,
      "paddingTop": 10,
    }
  }
  testID="base-screen"
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 251, 254, 1)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "marginBottom": 24,
        }
      }
    >
      <Text
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "writingDirection": "ltr",
            },
            [
              {
                "fontFamily": "System",
                "fontSize": 22,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 28,
              },
              [
                {
                  "color": "rgba(125, 82, 96, 1)",
                },
                {
                  "color": undefined,
                  "fontSize": 24,
                  "fontWeight": "700",
                  "marginBottom": 7,
                  "textAlign": "center",
                },
              ],
            ],
          ]
        }
      >
        
Thank you!
Your order has been placed successfully.
      
      </Text>
      <Text
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "writingDirection": "ltr",
            },
            [
              {
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "400",
                "letterSpacing": 0.25,
                "lineHeight": 20,
              },
              [
                {
                  "color": "rgba(125, 82, 96, 1)",
                },
                {
                  "color": undefined,
                  "fontSize": 16,
                },
              ],
            ],
          ]
        }
      >
        Redirecting to product list in 
        5
         second
        s
        ...
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": undefined,
          "borderRadius": 5,
          "height": 10,
          "marginBottom": 16,
          "overflow": "hidden",
          "width": "80%",
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "height": "100%",
            "width": "0%",
          }
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          colors={
            [
              undefined,
              undefined,
            ]
          }
          endPoint={
            [
              1,
              0,
            ]
          }
          startPoint={
            [
              0,
              0,
            ]
          }
          style={
            [
              {
                "flex": 1,
              },
              {},
            ]
          }
          testID="linear-gradient"
        />
        <View
          collapsable={false}
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": [
                {
                  "translateX": -100,
                },
              ],
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            colors={
              [
                16777215,
                1308622847,
                16777215,
              ]
            }
            endPoint={
              [
                1,
                0,
              ]
            }
            startPoint={
              [
                0,
                0,
              ]
            }
            style={
              [
                {
                  "flex": 1,
                },
                {},
              ]
            }
            testID="linear-gradient"
          />
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(98, 91, 113, 1)",
          "borderRadius": 20,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgba(98, 91, 113, 1)",
            "borderColor": "transparent",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 20,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      {
                        "color": "rgba(125, 82, 96, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      undefined,
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              Go to Products
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
